### PR TITLE
Reverse Engineer HasComputedColumnSql() as needed.

### PR DESCRIPTION
**Pull request by lajones**
Fix for #4944. For computed columns used `HasComputedColumnSql()` instead of `HasDefaultValueSql()`.

### CODE REVIEW

File: test/Microsoft.EntityFrameworkCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/AllFluentApi/SqlServerReverseEngineerTestE2EContext.expected [hunk 1]
```diff
@@ -271,14 +271,21 @@ namespace E2ETest.Namespace
                 entity.HasIndex(e => new { e.A, e.B })
                     .HasName("Test_PropertyConfiguration_Index");
 
+                entity.Property(e => e.ComputedDateTimeColumn)
+                    .HasColumnType("datetime")
+                    .HasComputedColumnSql("getdate()")
+                    .ValueGeneratedOnAddOrUpdate();
```
Comment (ErikEJ) [inline test/Microsoft.EntityFrameworkCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/AllFluentApi/SqlServerReverseEngineerTestE2EContext.expected]: GETDATE() is a default value, it is not computed on updates.... (unless you add some other RDBMS feature to do this)

Comment (lajones) [inline test/Microsoft.EntityFrameworkCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/AllFluentApi/SqlServerReverseEngineerTestE2EContext.expected]: Had a chat with @divega and we think this is a little different.

The definition of that column is `ComputedDateTimeColumn AS GETDATE()` i.e. it is a computed column, not a default value. We have a separate column to test a datetime2 column with a default value which is a computed expression (see the `WithDateDefaultExpression datetime2 NOT NULL DEFAULT(getdate())` column also on the `PropertyConfiguration` table). The 2 types of columns are treated differently in SQL Server metadata - the first is treated as a computed column (has an entry in `sys.computed_columns`); the second is not a computed column but does have a default value.

So, although technically you're right, the server does not compute a new value when a row is updated, in practice we keep the `ChangeTracker` up to date with updates by performing a select immediately after we've updated the data. Now on `SELECT` the value of this column will be updated, so the value stored in the `ChangeTracker` would likewise be updated. The closest we can come to this behavior is to assign it `ValueGeneratedOnAddOrUpdate()`.

Comment (ErikEJ) [inline test/Microsoft.EntityFrameworkCore.SqlServer.Design.FunctionalTests/ReverseEngineering/Expected/AllFluentApi/SqlServerReverseEngineerTestE2EContext.expected]: Got it, thanks!

File: src/Microsoft.EntityFrameworkCore.Relational.Design/Metadata/ColumnModel.cs [hunk 2]
```diff
@@ -21,6 +21,8 @@ public class ColumnModel : Annotatable
         [CanBeNull]
         public virtual string DefaultValue { get; [param: CanBeNull] set; }
 
+        [CanBeNull]
```
Comment (ajcvickers) [inline src/Microsoft.EntityFrameworkCore.Relational.Design/Metadata/ColumnModel.cs]: We decided in a design meeting some time back to not use these attributes for return values.

**Comment by ajcvickers** [comment]
:shipit:

File: src/Microsoft.EntityFrameworkCore.Relational.Design/Metadata/ColumnModel.cs (hunk 2 above)
Comment (lajones) [inline src/Microsoft.EntityFrameworkCore.Relational.Design/Metadata/ColumnModel.cs]: OK. Will remove (and similarly from the existing one above). Thx.

**Comment by lajones** [comment]
Found a few more `[CanBeNull]`s on return values. Removed them too and checked in with 2acd2f1.

**Pull request closed without merging**